app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -16,6 +16,8 @@ def index
                             client_id: @authorize_form.client_id,
                             errors: @authorize_form.errors.messages)
 
+      return create if already_allowed?
```

Comment:
Why do we need to submit the form again? If we know they are already allowed, can't we redirect to the `redirect_uri` defined in `service_providers.yml?`